sz-poc-offsite-2025/evm/sequencer-node/src/main.rs

```diff
@@ -0,0 +1,70 @@
+use evm_aggregator::Aggregator;
+use futures::TryStreamExt as _;
+use reth::{
+    api::{FullNodeTypes, NodePrimitives, NodeTypes},
+    cli::Cli,
+};
+use reth_ethereum::{
+    exex::{ExExContext, ExExEvent, ExExNotification},
+    node::{EthereumNode, api::FullNodeComponents},
+};
+use reth_tracing::tracing::info;
+
+async fn aggregate_block_txs<Node: FullNodeComponents>(
+    mut ctx: ExExContext<Node>,
+    mut aggregator: Aggregator,
+) -> eyre::Result<()>
+where
+    <<Node as FullNodeTypes>::Types as NodeTypes>::Primitives:
+        NodePrimitives<Block = reth_ethereum::Block>,
+{
+    while let Some(notification) = ctx.notifications.try_next().await? {
+        let ExExNotification::ChainCommitted { new } = &notification else {
```

Comment:
This ignores chain reorgs but the code at line 34 does not. It shouldn't be hard to support reorgs, but in case you're not interested, just could to `new.tip().num_hash()` instead of another call at line 34

Reply:
Yeah good catch, I fixed it here: https://github.com/logos-co/nomos-pocs/pull/59/commits/5eb976c9ae4f7446e8688d1a9b49ebbd1e30e959. I think if we go with this setup of a single validator running with PoA we can definitely ignore re-orgs, for the sake of PoC's simplicity.